Documentation/articles/domain_driven_design/concepts_and_value_objects.md

```diff
@@ -0,0 +1,114 @@
+
+# ConceptAs\<T>
+
+*ConceptAs\<T>* is intended as a direct replacement for where you would use a single primitive to represent
+a concept in the domain.  As well as implementing equality checks, and producing hash codes, *ConceptAs\<T>*
+allows an (almost) seamless transformation between the underlying primitive and the domain concept. This 
+proves to be very useful at the boundaries of your application, where you may be receiving primitives. It 
+can also aid in serialization and integration with other systems. *ConceptAs\<T>* implements an implicit 
+casting between the *ConceptAs\<T>* and the primitive *T*.
+
+A typical implementation of a ConceptAs might be
+
+```C#
+public class TelephoneNumber : ConceptAs<string>
+{
+    public static implicit operator TelephoneNumber(string value)
+    {
+        return new TelephoneNumber { Value = value };
+    }
+}
+```
+
+While the underlying ConceptAs<T> class can handle the implicit conversion to a string, it is not possible
+to implement the conversion to the specific type (i.e. TelephoneNumber) in a generic manner in the base class.
+
+It is **NOT RECOMMENDED** to enforce *correctness* of a ConceptAs\<T> by throwing an exception if there is 
+an attempt to create an instance in an invalid state.  It is the preferred strategy in Dolittle to capture
+invalid state through the mechanism of [input validation](./validation). This allows a more graceful and 
+informative handling of the invalid state.
+
+{{% notice tip %}}  
+Use of ConceptAs\<T> can lead to loss of type inference in e.g. lambda functions.  It can also require a
+little more work on the part of the developer when serializing.  We do not consider this too heavy a price
+to pay given the explicitness and expressiveness it brings to your domain.
+{{% /notice %}}  
+
+# Value\<T>
+
+*Value\<T>* provides basic value-object semantics by handling equality and production of hash codes.  When
+a class *T* inherits from *Value\<T>*, then equality and hash code will depend on the values of all public 
+properties.  In contrast to *ConceptAs\<T>*, *Value\<T>* is intended to group properties that belong together
+into a domain concept.  For example, a `Money : Value<Money>` might combine a monetary amount with a currency.
+A `Point : Value<Point>` might have properties for the X, Y and Z axis.
+
+It is *RECOMMENDED* and *ENCOURAGED* to use *ConceptAs\<T>* as properties within *Value\<T>*.  It is 
+*DISCOURAGED* but *NOT PROHIBITED* from using *Value\<T>* with just a single property.
+
+{{% notice tip %}}  
+Value objects within Dolittle consider the *System Type* to be integral to their identity.  While it is 
+permissible to inherit from a Value\<T> or ConceptAs\<T> that you have created, a type of the base class and 
+a type of subclass **WILL NOT** be considered equal, even if they have all properties identical.  A value
+object can only be equal to itself or another object of the same type.
+{{% /notice %}}  
+
```

Comment:
There's no IComparable<> support for Value<>.  We have no way of knowing that so the user has to do it themselves.